go/stats/prombackend/prombackend.go

```diff
@@ -0,0 +1,160 @@
+package prombackend
+
+import (
+	"net/http"
+	"strings"
+	"unicode"
+
+	prom "github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"vitess.io/vitess/go/stats"
+)
+
+// PromBackend implements PullBackend using Prometheus as the backing metrics storage.
+type PromBackend struct {
+	namespace string
+}
+
+// Init initializes the Prometheus backend with the given namespace.
+func Init(namespace string) {
+	http.Handle("/metrics", promhttp.Handler())
+	promBackend := &PromBackend{namespace: namespace}
+	stats.RegisterPullBackendImpl("prom", promBackend)
```

Comment:
This should just call `stats.Register` with that new function that can call the various pull functions.